src/sheets/updateItems.js

```diff
@@ -1,38 +1,61 @@
-import logger from '../logger'
-import {getValues, batchUpdateValues} from './sheets.js'
+import {getFieldIndexMap, getValues, batchUpdateValues} from './sheets.js'
+import {getColumn} from './utils'
 import {sheets} from './constants'
 
 export async function updateItems(
   isCompanyItem,
   items,
-  getCompanyItemData = () => null,
-  getPersonalItemData = () => null,
+  getItemData,
 ) {
   const sheet = isCompanyItem ? sheets.companyOrders : sheets.personalOrders
 
-  const itemsMap = new Map(items.map((item) => [item.id, item]))
+  const [fieldIndexMap, itemIdsFromSheet] = await Promise.all([
+    getFieldIndexMap(sheet.name, sheet.fieldsRow),
+    getValues(sheet.idRange),
+  ])
 
-  const itemIdsFromSheet = await getValues(sheet.idRange)
+  const idToRowsMap = {}
 
-  const data = itemIdsFromSheet.reduce((result, itemId, index) => {
-    if (itemsMap.has(itemId[0])) {
-      const rowIndex = index + sheet.rowOffset + 1
+  itemIdsFromSheet.forEach(([itemId], index) => {
+    if (itemId) {
+      if (!idToRowsMap[itemId]) {
+        idToRowsMap[itemId] = []
```

Comment:
Why does this need to be an array? Just in case that multiple items have the same id? Can that happen?

Reply:
Yes, it's for the case when multiple items have the same id. It can theoretically happen, because Daska sometimes adds items to the sheet manually. I warned her to use some prefix for manually added items